src/network/C4Network2.cpp

```diff
@@ -909,26 +928,46 @@ bool C4Network2::HandlePuncherPacket(C4NetpuncherPacket::uptr pkt)
 		case PID_Puncher_AssID:
 			if (isHost())
 			{
-				NetpuncherGameID = GETPKT(AssID)->GetID();
+				getNetpuncherGameID(family) = GETPKT(AssID)->GetID();
 				InvalidateReference();
 			}
 			else
 			{
 				// While we don't need the ID as a client, this nicely serves as the signal that we can start using the netpuncher
-				if (Status.getState() == GS_Init && getNetpuncherGameID())
-					NetIO.SendPuncherPacket(C4NetpuncherPacketSReq(getNetpuncherGameID()));
+				if (Status.getState() == GS_Init && getNetpuncherGameID(family))
+					NetIO.SendPuncherPacket(C4NetpuncherPacketSReq(getNetpuncherGameID(family)), family);
 			}
 			return true;
 		default: return false;
 	}
 }
 
+C4NetpuncherID_t& C4Network2::getNetpuncherGameID(C4NetIO::HostAddress::AddressFamily family)
+{
+    switch (family)
+    {
+    case C4NetIO::HostAddress::IPv4: return NetpuncherGameID.v4;
+    case C4NetIO::HostAddress::IPv6: return NetpuncherGameID.v6;
+    default: assert(false);
```

Comment:
- Can you replace this with an assert that outputs a more useful message when it fails? E.g. `assert("Unexpected address family");` or something like that.
- This case still needs to return a value since in release builds we disable assertions and control will reach the end of the function.